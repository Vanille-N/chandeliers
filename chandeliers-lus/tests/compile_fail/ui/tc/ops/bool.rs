chandeliers_lus::decl! {
    node foo(i : int) returns (b : bool);
    let
        b = if i then 0 else 1;
    tel
}

chandeliers_lus::decl! {
    node bar(i : int) returns ();
    let
        assert i;
    tel
}

fn main() {}
