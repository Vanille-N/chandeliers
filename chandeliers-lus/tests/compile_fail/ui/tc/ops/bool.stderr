error: The condition of if should be of type bool
 --> tests/compile_fail/ui/tc/ops/bool.rs:6:13
  |
6 |         b = if i then 0 else 1;
  |             ^^^^^^^^^^^^^^^^^^
  |
note: The argument is found to be of type int
 --> tests/compile_fail/ui/tc/ops/bool.rs:6:16
  |
6 |         b = if i then 0 else 1;
  |                ^

error: The argument of assert should be of type bool
  --> tests/compile_fail/ui/tc/ops/bool.rs:13:9
   |
13 |         assert i;
   |         ^^^^^^^^
   |
note: The argument is found to be of type int
  --> tests/compile_fail/ui/tc/ops/bool.rs:13:16
   |
13 |         assert i;
   |                ^
